mkosi/distributions/gentoo.py

```diff
@@ -10,189 +8,98 @@
 from textwrap import dedent
 
 from mkosi.architecture import Architecture
+from mkosi.config import ConfigFeature
 from mkosi.distributions import DistributionInstaller
-from mkosi.install import copy_path, flock
+from mkosi.install import copy_path
 from mkosi.log import ARG_DEBUG, complete_step, die, log_step
 from mkosi.remove import unlink_try_hard
 from mkosi.run import run, run_workspace_command
 from mkosi.state import MkosiState
+from mkosi.types import PathString
 
 
 def invoke_emerge(
     state: MkosiState,
+    sysroot: Path,
     pkgs: Sequence[str] = (),
     actions: Sequence[str] = (),
     opts: Sequence[str] = (),
+    env: dict[str, str] = {},
 ) -> None:
-    jobs = os.cpu_count() or 1
-    emerge_default_opts = [
+    # This is the mount-point inside our sysroot where we mount root
+    bwrap: list[PathString] = [
+        "--bind", state.root, "/tmp/mkosi-root",
+        "--bind", state.cache_dir / "binpkgs", "/var/cache/binpkgs",
+        "--bind", state.cache_dir / "distfiles", "/var/cache/distfiles",
+        "--bind", state.cache_dir / "repos", "/var/db/repos",
+    ]
+    root = Path("/tmp/mkosi-root")
+
+    cmd = [
+        "emerge",
+        *pkgs,
         "--buildpkg=y",
         "--usepkg=y",
         "--keep-going=y",
-        f"--jobs={jobs}",
-        f"--load-average={jobs+1}",
+        "--jobs",
+        "--load-average",
         "--nospinner",
+        *([f"--root={root}"] if root != sysroot else []),
```

Comment:
When will: `root == sysroot` ever be true?

Reply:
you're right, in this form it will never be.